### ISSUE TITLE
Parameter includePlatform in Download-Artifacts (sometimes) not working 

### ISSUE THREAD

**Issue by NAVRockClimber** [microsoft/navcontainerhelper#1137]
Hi,

if I try to download the artifacts including the plattform artifacts the download seems to work but after extracting the zip files no plattform folder is present in my cache directory. See screenshot in the screenshot section below.

This issue to occurs in the versions 0.7.0.16 and 0.7.0.17. 
It seems you switched in these version from the built-in unzip command to 7zip. I suspect 7zip not finished at the point where you try to rename the folder.

Command executed:
```
Flush-ContainerHelperCache
$bcartifacturl = Get-BCArtifactUrl -country de -type OnPrem
Download-Artifacts -artifactUrl $bcartifacturl -includePlatform
```

**Full output of scripts**
```
Downloading application artifact /onprem/16.3.14085.14238/de
Downloading C:\Users\<MyUser>\AppData\Local\Temp\61d79ed2-a958-4115-a465-6530894f22b9.zip
Unpacking application artifact to tmp folder using 7zip
c:\bcartifacts.cache\onprem\16.3.14085.14238\de
Downloading platform artifact /onprem/16.3.14085.14238/platform
Downloading C:\Users\<MyUser>\AppData\Local\Temp\8f80213a-9927-47dc-b45e-f27db91d7efe.zip
Unpacking platform artifact to tmp folder using 7zip
c:\bcartifacts.cache\onprem\16.3.14085.14238\platform
```

**Screenshots**
![Untitled](https://user-images.githubusercontent.com/37535503/87775123-dd473d00-c825-11ea-8779-ed7e743a3239.png)

**Additional context**
- does it happen all the time? Unfortunately this seems to be an issue only on out build servers and a few other machines
- did it use to work? This worked till version 0.7.0.15

**Results from personal investigation so far yet**

I suspect after extracting the plattform zip file 7zip is not finished yet or a file handle is still open.

What I have done:
Removed on the Rename-Item command for the plattform tempfolder the paramter -ErrorAction SilentlyContinue

![2020_07_17_12_19_48_Window](https://user-images.githubusercontent.com/37535503/87777486-dc180f00-c829-11ea-8881-cc8c9a780dec.png)

Repeated command sequence from above:
```
Flush-ContainerHelperCache
$bcartifacturl = Get-BCArtifactUrl -country de -type OnPrem
Download-Artifacts -artifactUrl $bcartifacturl -includePlatform
```
New output of scripts:
```
Downloading application artifact /onprem/16.3.14085.14238/de
Downloading C:\Users\<MyUser>\AppData\Local\Temp\eea95a24-1dad-42cf-b429-e3da3d4f5a6a.zip
Unpacking application artifact to tmp folder using 7zip
c:\bcartifacts.cache\onprem\16.3.14085.14238\de
Downloading platform artifact /onprem/16.3.14085.14238/platform
Downloading C:\Users\<MyUser>\AppData\Local\Temp\93b1f2f8-ef4f-4b02-8484-f921bf0bc49c.zip
Unpacking platform artifact to tmp folder using 7zip
Rename-Item : Access to the path 'C:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305858774199506' is denied.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.17\Misc\Download-Artifacts.ps1:175 char:29
+ ...             Rename-Item -Path "$tmpFolder" -NewName ([System.IO.Path] ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (C:\bcartifacts....305858774199506:String) [Rename-Item], IOException
    + FullyQualifiedErrorId : RenameItemIOError,Microsoft.PowerShell.Commands.RenameItemCommand

```

**Comment by freddydk** [comment]
Seems like I need a retry algorithm on that rename.
Could you try to download https://raw.githubusercontent.com/microsoft/navcontainerhelper/dev/Misc/Download-Artifacts.ps1 and override the file on your machines to see whether this solves the issue?

if, then next Containerhelper will include this fix.

/Freddy

**Comment by NAVRockClimber** [comment]
Thanks for the fast response.

Your fix seems to work. 

My Output:
```
Downloading application artifact /onprem/16.3.14085.14238/de
Downloading C:\Users\<MyUser>\AppData\Local\Temp\50912483-afb6-49b7-a6e7-6631292e8045.zip
Unpacking application artifact to tmp folder using 7zip
c:\bcartifacts.cache\onprem\16.3.14085.14238\de
Downloading platform artifact /onprem/16.3.14085.14238/platform
Downloading C:\Users\<MyUser>\AppData\Local\Temp\8d5b79cb-4fef-4965-a3dc-c577182d5df6.zip
Unpacking platform artifact to tmp folder using 7zip
Could not rename 'c:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305938069687808' retrying in 5 seconds.
Retrying...
Could not rename 'c:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305938069687808' retrying in 5 seconds.
Retrying...
Could not rename 'c:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305938069687808' retrying in 5 seconds.
Retrying...
Could not rename 'c:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305938069687808' retrying in 5 seconds.
Retrying...
Could not rename 'c:\bcartifacts.cache\onprem\16.3.14085.14238\tmp637305938069687808' retrying in 5 seconds.
Retrying...
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\onprem\16.3.14085.14238\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\onprem\16.3.14085.14238\platform\Prerequisite Components\Microsoft Report Viewer 2015\ReportViewer.msi
Downloading c:\bcartifacts.cache\onprem\16.3.14085.14238\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\onprem\16.3.14085.14238\platform\Prerequisite Components\Microsoft Report Viewer 2015\SQLSysClrTypes.msi
Downloading c:\bcartifacts.cache\onprem\16.3.14085.14238\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
c:\bcartifacts.cache\onprem\16.3.14085.14238\platform
```

**Comment by freddydk** [comment]
Fixed in 0.7.0.18
